### PR TITLE
Switch Master references to Main as master-branch of nav2 is deleted

### PR DESCRIPTION
**Pull request by gramss**
Open questions: 
* Is Ros2 already on Main? (first 2 commits) - but only text
* ClearCostmap has some Master references that must be changed before docs change (master_grid)

+ adding dependency instructions for this README
+ small fix displaying ``--no-ros2`` correctly

 Did build successfully on my machine.

### CODE REVIEW

File: build_instructions/index.rst [hunk 1]
```diff
@@ -48,7 +48,7 @@ Note: You need to change ``--rosdistro`` to the selected ROS 2 distribution name
 
 .. rst-class:: content-collapse
 
-Quickstart Build Master
+Quickstart Build  Main
```
Comment (SteveMacenski) [inline build_instructions/index.rst:51]: You need to make sure the bars below `====` match the width of the header text

File: build_instructions/index.rst [hunk 2]
```diff
@@ -82,23 +82,23 @@ Options
...
 
-- `--no-ros2` This skips downloading and building the ROS 2 release. Instead it uses the binary packages and ``setup.sh`` installed in ``/opt/ros/<ros2-distro>``
+- ``--no-ros2`` This skips downloading and building the ROS 2 release. Instead it uses the binary packages and ``setup.sh`` installed in ``/opt/ros/<ros2-distro>``
 - ``--download-only`` This skips the build steps
 
 
 .. rst-class:: content-collapse
 
-Manually Build Master
+Manually Build  Main
```
Comment (SteveMacenski) [inline build_instructions/index.rst:91]: Same with header bars here and one below

File: build_instructions/index.rst [hunk 3]
```diff
@@ -117,24 +117,24 @@ Then, use ``vcs`` to clone the repos and versions in it into a workspace.
...
   mkdir -p ~/nav2_depend_ws/src
   cd ~/nav2_depend_ws
-  wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/ros2_dependencies.repos
+  wget https://raw.githubusercontent.com/ros-planning/navigation2/main/tools/ros2_dependencies.repos
   vcs import src < ros2_dependencies.repos
   rosdep install -y -r -q --from-paths src --ignore-src --rosdistro <ros2-distro>
   colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=Release
 
-Build Navigation2 Master
+Build Navigation2  Main
```
Comment (SteveMacenski) [inline build_instructions/index.rst:125]: header bars

File: build_instructions/index.rst (hunk 1 above)
Comment (gramss) [reply build_instructions/index.rst]: oh thanks. did not see that as I was just mass replacing master with main and checked the context in the individual line
Should be fixed now

**Pull request merged**